### DISCUSSION TITLE
Is it possible to upload .wav files to DANDI archive

### DISCUSSION THREAD

**Discussion by magland** [dandi/helpdesk#117] [General]
Is it possible to upload .wav files to DANDI archive? If so, what should I do in the `dandi upload` command in the CLI?

**Comment by CodyCBakerPhD** [comment]
Not currently possible - please include internally to the NWB file

Check out the [NeuroConv AudioInterface](https://neuroconv.readthedocs.io/en/main/conversion_examples_gallery/behavior/audio.html) to do this easily

**Comment by magland** [comment]
Thanks @CodyCBakerPhD 

I've got 8 .wav files for one experiment, for 8 microphones. Is there a way to modify that example to get them all into a single NWB?

**Reply by CodyCBakerPhD** [reply]
- instantiate a separate AudioInterface by looping over the files, adding them to a common list

```python
all_audio_file_paths = [".../first_audio_file.wav", ...]
data_interfaces = [AudioInterface(file_paths=[audio_file_path]) for audio_file_path in all_audio_file_paths]
```

- Use a `ConverterPipe` as seen in [this example](https://neuroconv.readthedocs.io/en/main/conversion_examples_gallery/combinations/spikeglx_and_phy.html) to glue them all together

- If this doesn't turn out as expected raise an issue on NeuroConv. We've not seen a multi-microphone example before so might need to modify one or two things to get the object names not colliding

**Reply by magland** [reply]
Thanks. As you anticipated there are name collisions. So I opened an issue https://github.com/catalystneuro/neuroconv/issues/637

**Comment by yarikoptic** [comment]
I wonder if the situation is similar to external videos. Is pointing to external files (ideally not uncomressed .wavs) supported in NWB for audio files?

**Reply by satra** [reply]
and if included in nwb, i hope it's not uncompressed, but compressed with wavpack. if that's the case, at least in case of audio compression levels should be very good and comparable.

**Reply by CodyCBakerPhD** [reply]
> Is pointing to external files (ideally not uncomressed .wavs) supported in NWB for audio files?

Not currently, the internal storage methods have been good enough compression-wise to not need it

The only type of compression we can't get close to when writing internally is the video codecs (usually 10-16 to 1 compression ratio on videos of natural behavior)



> and if included in nwb, i hope it's not uncompressed

If Zarr, yes Wavpack codec is supported

If HDF5, I don't know of a Wavpack `hdfplugin` so we have used GZIP and that has produced satisfactory compression ratios in the past
